src/rust/engine/hashing/src/lib.rs

```diff
@@ -154,6 +155,14 @@ impl<'de> Deserialize<'de> for Fingerprint {
   }
 }
 
+impl FromStr for Fingerprint {
```

Comment:
Worth also adding a `TryFrom` impl?

Reply:
Done in https://github.com/pantsbuild/pants/pull/11306